core/shared/src/main/scala/cats/effect/unsafe/IOLocals.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020-2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect
+package unsafe
+
+import IOLocalsConstants.ioLocalPropagation
+
+object IOLocals {
+
+  /**
+   * `true` if IOLocal propagation is enabled
+   */
+  def arePropagating: Boolean = ioLocalPropagation
```

Comment:
```suggestion
  def isPropagating: Boolean = ioLocalPropagation
```

Slightly more idiomatic naming.

Reply:
Previously bikeshed in https://github.com/typelevel/cats-effect/pull/3636#discussion_r1340726643 but I switched to `isPropagating`.